content/en/observability_pipelines/guide/custom-metrics-governance.md

```diff
@@ -0,0 +1,213 @@
+    inputs:
+      - dd_agent.metrics
+    source: |2+
+
+           # list of tags to keep
+           tags_allowlist = ["tag 1", "tag 2", "tag 3", "tag 4"]
+
+           # filter and drop any tags that don't match the tags in tags_to_keep
+           .tags = [filter(.tags) -> |_index, value| { includes(tags_allowlist, value)}
+```
+
+### Solution 3: Define a blocklist array of tags to drop
+
+In the case where you know which tags you want to drop, use the `filter()` function to define a blocklist of those tags. This is done by adding `!` to `includes(tags_blocklist, value)` from Solution 2.
+
+```yaml
+transforms:
+  drop_excluded_tags:
+    type: remap
+    inputs:
+      - dd_agent.metrics
+    source: |2+
+
+           # list of tags to keep
+           tags_blocklist = ["tag 1", "tag 2", "tag 3", "tag 4"]
+
+           # filter and drop any tags that don't match the tags in tags_to_keep
+           .tags = [filter(.tags) -> |_index, value| { !includes(tags_blocklist, value)}
+```
+
+### Solution 4: Define an allowlist of valid tags in a Reference Table
+
+In cases when you have a specific list of valid tags, you can use Observability Pipelines' `get_enrichment_tables` and `find_enrichment_tables` functions. This allows you to reference an enrichment table in `csv` format in Observability Pipelines. `csv` is the only supported file format. 
+
+For this example, a `csv` file named `valid_tags.csv` contains the following valid tags:
+
+```
+tag_name
+"tag 1"
+"tag 2"
+"tag 3"
+"tag 4"
+```
+
+To reference it as table, the following configuration must be set for the file `valid_tags.csv`.
+
+```
+enrichment_tables:
+  valid_tag_table:
+    type: file
+    file:
+      path: /etc/vector/valid_tags.csv
+      encoding:
+        type: csv
+    schema:
+      tag_name: string
+```
+
+Once the `enrichment_tables` field is set, you can drop tags not included in the `valid_tags.csv` file using the following configuration:
+
+
+```yaml
+transforms:
+  drop_excluded_tags:
+    type: remap
+    inputs:
+      - dd_agent.metrics
+    source: |2+
+           # filter and drop any tags that don't match the tags in tags_to_keep
+           .tags = [filter(.tags) -> |_index, value| {!is_empty(find_enrichment_table_records(valid_tag_table, value))}
+```
+
+## Prevent cardinality spikes
+
+### Problem
+
+Your overall volume of custom metrics is dependent on the number of tag value combinations submitted to Datadog. See [Custom Metrics Billing][8] on count custom metrics are counted. It is possible for a particular tag key's cardinality to unintentionally spike. For example, the `path` tag usually has 100 unique tag values, but then suddenly spikes to 100,000 unique values. This sudden increase in tag cardinality can cause an increase in your overall custom metrics volumes. It can also potentially cause you to exceed your account allotment for custom metrics. To avoid this situation, you can set up rules in Observability Pipelines to prevent cardinality spikes.
```

Comment:
```suggestion
Your overall volume of custom metrics is dependent on the number of tag value combinations submitted to Datadog. See [Custom Metrics Billing][8] to learn how custom metrics are counted. It is possible for a particular tag key's cardinality to unintentionally spike. For example, the `path` tag usually has 100 unique tag values, but then suddenly spikes to 100,000 unique values. This sudden increase in tag cardinality can cause an increase in your overall custom metrics volumes. It can also potentially cause you to exceed your account allotment for custom metrics. To avoid this situation, you can set up rules in Observability Pipelines to prevent cardinality spikes.
```